packages/google_sign_in/google_sign_in_web/lib/google_sign_in_web.dart

```diff
@@ -27,24 +44,56 @@ class GoogleSignInPlugin extends GoogleSignInPlatform {
   /// Constructs the plugin immediately and begins initializing it in the
   /// background.
   ///
-  /// The plugin is completely initialized when [initialized] completed.
-  GoogleSignInPlugin({@visibleForTesting bool debugOverrideLoader = false}) {
+  /// For tests, the plugin can skip its loading process with [debugOverrideLoader],
+  /// and the implementation of the underlying GIS SDK client through [debugOverrideGisSdkClient].
+  GoogleSignInPlugin({
+    @visibleForTesting
+        bool debugOverrideLoader = false,
+    @visibleForTesting
+        GisSdkClient? debugOverrideGisSdkClient,
+    @visibleForTesting
+        StreamController<GoogleSignInUserData?>?
+            debugOverrideUserDataController,
+  })  : _gisSdkClient = debugOverrideGisSdkClient,
+        _userDataController = debugOverrideUserDataController ??
+            StreamController<GoogleSignInUserData?>.broadcast() {
     autoDetectedClientId = html
         .querySelector(clientIdMetaSelector)
         ?.getAttribute(clientIdAttributeName);
 
+    _registerButtonFactory();
+
     if (debugOverrideLoader) {
       _jsSdkLoadedFuture = Future<bool>.value(true);
     } else {
       _jsSdkLoadedFuture = loader.loadWebSdk();
     }
   }
 
+  // A future that completes when the JS loader is done.
   late Future<void> _jsSdkLoadedFuture;
+  // A future that completes when the `init` call is done.
+  final Completer<void> _initDone = Completer<void>();
+  // A (synchronous) marker to assert that `init` has been called.
   bool _isInitCalled = false;
```

Comment:
Optional: would it make sense to remove this and make `_initDone` optional, with a null future indicating that it hasn't been called, so there's only one thing tracking the init flow?

Reply:
Actually, yes, it works Just Fine. Thanks for the suggestion!